### PR TITLE
WIP - Esri portal connector

### PR DESCRIPTION
**Pull request by rowanwins**
### What this PR does
This PR adds a new connector for harvesting data from an Esri Portal via [their Rest API](https://developers.arcgis.com/rest/users-groups-and-items/working-with-users-groups-and-items.htm).

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable
-   [ ] I've updated CHANGES.md with what I changed.
-   [ ] I've linked this PR to an issue in ZenHub (core dev team only)

### CODE REVIEW

File: magda-esri-portal-connector/src/EsriPortalUrlBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        return this.apiBaseUrl
+            .clone()
+            .segment("portals/self")
+            .addSearch({ f: "pjson" })
+            .toString();
+    }
+
+    // https://someportal/arcgis/sharing/rest/content/items/ea532dd8876b4646aa2ab49533e458aa?f=pjson
+    public getContentItemUrl(id: string): string {
+        // TODO Sometimes getting an id of undefined...
```
Comment (rowanwins) [inline magda-esri-portal-connector/src/EsriPortalUrlBuilder.ts]: This is where seems to be throwing the error when trying to create a distribution source

**Comment by rowanwins** [comment]
@kring I think this is now working reasonably well

In terms of the initial crawl of data I'm restricting the initial harvesting of data to `Map Service` and `Feature Service` data types. If you do a general query you get back a squillion irrelevant things like Apps, Tools etc etc.

When setting up the data and distributions I'm crawling the individual layers and adding them as distributions. For example if an item in portal is listed as `/some/MapService` then that becomes the dataset, and I then create distributions for `/some/MapService`, `/some/MapService/0`, `/some/MapService/1` etc etc 

DatasetWithDistributions

Some outstanding questions in my mind...
- how do we pass auth through to a connector (eg so we can access the private content), as it stands the search is only retrieving public content.
- Do we want to add to the data type search `Scene Service` so they get pulled into magda as well even through we don't quite support them in Terria at the moment?

**Comment by rowanwins** [comment]
travis-ci/push is unhappy because there are no tests at the moment

**Comment by mwu2018** [comment]
Superseded by PR #2483

**Pull request closed without merging**